### ISSUE TITLE
Add a new line before file ending when saving

### ISSUE THREAD

**Issue by zuphilip** [zotero/scaffold#37]
It looks that during saving a translator with Scaffold there is no new line at the file ending. However, files should end with a newline, as this is [part of the POSIX standard](http://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap03.html#tag_03_206) and most tools as well as the editors will expect that. Actually, it is quite frequent that in commits for translators this difference shows up, e.g. most recent https://github.com/zotero/translators/commit/16e77e81c046c11abc9a6beea668a9eab5d2df74#diff-28f549f8b27210b22f79c52c0eac9d73L894 . HT @kba

![newline](https://cloud.githubusercontent.com/assets/5199995/22200685/62cf8c62-e160-11e6-8144-aea1ce34e5b4.png)

Can we change the saving in Scaffold such that it will also include a newline before the file ending?

**Comment by dstillman** [comment]
This will be in the next Zotero 5.0 beta, build 117.

**Comment by zuphilip** [comment]
Okay, that is good to hear. I don't know how scaffold and this zotero code is related but we could also additionally correct this already on the scaffold level (there is no reason to systematically not doing it in scaffold). I would suggest to [change these lines](https://github.com/zotero/scaffold/blob/df467621397cfbc0c0a797fc8946571673b95751/src/chrome/content/scaffold/scaffold.js#L644-L645) to:
```diff
- var code = "/** BEGIN TEST CASES **/\nvar testCases = "
+ var code = "\n/** BEGIN TEST CASES **/\nvar testCases = "
- 	testString + "\n/** END TEST CASES **/";
+ 	testString + "\n/** END TEST CASES **/\n";
```
Also in the beginning there should be IMO a newline, because otherwise this is much more difficult to parse (e.g. in the travis check).

Can I prepare such a PR here?

**Comment by dstillman** [comment]
There's no point — Scaffold saves through Zotero, the code is already updated to only work with 5.0, and this is fixed for 5.0.

**Comment by dstillman** [comment]
Oh, didn't see the part about the beginning newline — we can fix that, then.
